src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java

```diff
@@ -0,0 +1,57 @@
+package cat.udl.eps.softarch.academicrecruit.steps;
+
+import cat.udl.eps.softarch.academicrecruit.domain.Candidate;
+import cat.udl.eps.softarch.academicrecruit.domain.Participant;
+import cat.udl.eps.softarch.academicrecruit.repository.CandidateRepository;
+import cat.udl.eps.softarch.academicrecruit.repository.ParticipantRepository;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.When;
+import org.json.JSONObject;
+import org.springframework.http.MediaType;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.hamcrest.Matchers.is;
+
+public class UpdateCandidateStepDefs {
+
+    final StepDefs stepDefs;
+    final CandidateRepository candidateRepository;
+    private String newResourceUri;
+
+    public UpdateCandidateStepDefs(StepDefs stepDefs, CandidateRepository candidateRepository) {
+        this.stepDefs = stepDefs;
+        this.candidateRepository = candidateRepository;
+    }
+
+
+    @When("I change the name of the candidate with id {string} to {string}")
+    public void iChangeNameOfCandidateTo(String id, String name) throws Throwable {
+
+        Candidate candidate = new Candidate();
+        candidate.setId(Long.valueOf(id));
+        candidate.setName(name);
+
+        newResourceUri = "/candidates/"+ id;
+
+        stepDefs.result = stepDefs.mockMvc.perform(
```

Comment:
You should create the existing candidate in a separate step (Given there is a candidate with name "oldname"...) and then update the name with a different one in this step